### PR TITLE
Properly quote %sources and %patches

### PR DESCRIPTION
**Pull request by ffesti**
by enclosing them in single quotes (').
Replace sigle quotes within with '\'' as needed for sh

Resolves: #1445

### CODE REVIEW

File: macros.in [hunk 1]
```diff
@@ -1054,8 +1054,10 @@ package or when debugging this package.\
   install
 
 #------------------------------------------------------------------------------
-%patches %{lua: for i, p in ipairs(patches) do print(p.." ") end}
-%sources %{lua: for i, s in ipairs(sources) do print(s.." ") end}
+%patches %{lua: for i, p in ipairs(patches) do \
+  print("'"..string.gsub(p, "'", "'\\\\''").."'".." ") end}
+%sources %{lua: for i, s in ipairs(sources) do \
+  print("'"..string.gsub(s, "'", "'\\\\''").."'".." ") end}
```
Comment (DemiMarie) [inline macros.in:1060]:
```suggestion
  print("'"..string.gsub(p, "'", "'\\\\''").."' ") end}
%sources %{lua: for i, s in ipairs(sources) do \
  print("'"..string.gsub(s, "'", "'\\\\''").."' ") end}
```

**Comment by pmatilai** [comment]
Adding the single quotes seems like a no-brainer on the outset, but when you need to escape them, it opens a different can of worms: macro use is not limited to shell, only there's no way to tell (rhymes nicely, no? :laughing: ) and can occur outside the spec as well (the number of backslashes seems to suggest a backslash-eating spec-context at work, didn't check though).

We of course do have numerous shell-only macros, but I'm not sure %patches and %sources are such by definition. At the very least, they are not the only place where quoting and consequent escaping might be needed - eg, what about single %{PATCH5} or %{SOURCE1} with funky characters in them?
So it seems to me we should have a generic helper macro for quoting (with escapes) for the shell.
Perhaps %patches and %sources deserve/need to be shell-only, but I'm not entirely convinced this is a safe assumption.

**Comment by ffesti** [comment]
Not sure if we can do a quoting macro. This patch has the benefit of running in lua and though work on the original content of the file name. A macro will always see the macro expanded version of the parameter. Not sure if there is a way around that with the current macro system - especially as one needs to expand `%{PATCH5}` or what every hold the data in question at some point.

**Comment by ffesti** [comment]
Yes, we need quoting at more places. Reopened #1644.

**Comment by pmatilai** [comment]
> A macro will always see the macro expanded version of the parameter. Not sure if there is a way around that with the current macro system - especially as one needs to expand %{PATCH5} or what every hold the data in question at some point.

Macro arguments are expanded yes, but you need that anyway to get at that %{PATCH5}. Built-in's and Lua macros can control what happens beyond that. There's %{quote:...} for example already, so why not %{shquote:...}. Hmm, actually there's also :shescape query formatter already...

**Comment by pmatilai** [comment]
See #1783

**Comment by ffesti** [comment]
I'll adjust this to use the new macro as soon as it is upstream.

**Comment by DemiMarie** [comment]
> See #1783 (sorry, this was way more fun than what I'm supposed to be doing...)

what is that?

**Comment by ffesti** [comment]
OK; turns out I already did that in September but did not push the changes here... If I only could remember why.

I also added a patch to quote patch names in %autosetup.

File: macros.in [hunk 2]
```diff
@@ -1259,7 +1261,7 @@ end
 for i, a in ipairs(arg) do
     local p = bynum[tonumber(a)]
     if p then
-        print(rpm.expand("%__apply_patch -m %{basename:"..p.."}  "..options..p.." "..i.."\\n"))
+        print(rpm.expand("%__apply_patch -m %{shescape:%{basename:"..p.."}}  "..options..p.." "..i.."\\n"))
```
[the remark this answers is not in the document]
Comment (pmatilai) [reply macros.in:1264]: (a testcase would help catching such details...)

**Comment by ffesti** [comment]
Yeah, guess there was some reason - like the second patch not working at all.

**Comment by dmnks** [comment]
LGTM, thanks!

**Pull request merged**